python-sdk/src/astro/databases/base.py

```diff
@@ -422,6 +423,18 @@ def fetch_all_rows(self, table: BaseTable, row_limit: int = -1) -> list:
         response = self.run_sql(statement)
         return response.fetchall()  # type: ignore
 
+    @staticmethod
+    def check_for_minio_connection(input_file: File) -> bool:
+        """Returns True if minio is passed in extras in connections."""
+        if input_file.location.location_type == FileLocation.S3:
+            try:
+                conn = input_file.location.hook.get_connection(input_file.conn_id)
+            except AirflowNotFoundException as exe:
+                logging.warning(exe)
+                return False
+            return bool(conn.extra_dejson.get("minio"))
```

Comment:
What is the advantage of this approach as opposed to identifying automagically if the object store is minio, for instance, using something like:
```
requests.get(https://minio.example.net:9000/minio/health/live)
```
https://min.io/docs/minio/linux/operations/monitoring/healthcheck-probe.html

Reply:
@tatiana I am okay with doing this change as well but I feel that this is an external URL which can be changed by minio. And there can be the scenarios where health check URL can be publicly pointing to a different port (or using reverse proxy depending on client setup).

With the current approach, the user only would need to add `minio=True` as part of the connection.

Reply:
@ sunank200 since this URL path directly affects production deployment (e.g. health checks and probes), changes are unlikely.

Regarding the port / reverse proxy - wouldn't those affect the overall Airflow connection endpoint the same way?

With the current approach (`minio=True`), we are adding overhead to end-users (who could not realise they needed to add this property to the connection, leading to the Python SDK incorrectly trying to use a native S3 transfer, when it shouldn't).

Reply:
I agree with @tatiana, we should not add another extra to the connection. The goal of tools such as MinIO and localstack is to provide the same interface. If we really need to check whether it is s3 or not, we should check the endpoint url as this is (and should be) the only differentiation.

For example, take a look at this guide.

More specifically, in the "Configuring Logs" section:
> In Extras, let's set the URL to our local MinIO deployment with the following syntax
> { "endpoint_url": "http://<ip>:9000" }

The endpoint url is what aws let us overwrite. localstack is using the same approach.

**So I would suggest we know that it is not native s3 if the conn type is aws s3 but the endpoint URL is defined** or the default s3 endpoint url is used if there is any.

Reply:
@tatiana Made changes according to healtcheck approach.

Reply:
@utkarsharma2 why do we need to get the health state? Even if minio is not up & running, we can detect if it is aws s3 or something else. Please see my comment above.

Reply:
@feluelle What will happen if the user runs minio server on some other port?

Reply:
@feluelle Should we just check for the presence of a port in the endpoint_url? because the user can host it locally(or on any server) and on any port, right?

Reply:
We check whether the endpoint url is set at all. Why would a user set the endpoint url if they are using native s3?

Reply:
There is a use case for this - https://docs.aws.amazon.com/AmazonS3/latest/userguide/WebsiteEndpoints.html

Reply:
That is the wrong "endpoints", see [this](https://docs.aws.amazon.com/general/latest/gr/rande.html).

The endpoint URL usually will be created from the information you provide in your aws connfig (or airflow connection). The only time (I can think of) you actually need it, is when you use a different service such as MinIO or localstack.

However, if our users are reporting issues about that (using custom endpoint which are targeting real aws services), we can think about it again. WDYT?

Reply:
@feluelle Make sense, updated code accordingly.